### PR TITLE
mender-shell Yocto updates: new variable and config functionality

### PR DESCRIPTION
**Pull request by lluiscampos**

### CODE REVIEW

**Comment by lluiscampos** [comment]
Implementation in https://github.com/mendersoftware/meta-mender/pull/1228

File: 05.System-updates-Yocto-Project/05.Customize-Mender/docs.md [hunk 1]
```diff
@@ -222,8 +222,24 @@ IMAGE_INSTALL_append = " mender-shell"
...
-augment the `mender-client` recipe with the new configuration. For example, create a `mender-client_%.bbappend` file in your layer, and add this:
+Mender-shell provides several [configuration
+options](../../03.Add-ons/01.Remote-Terminal/30.Mender-shell-configuration-file/50.Mender-shell-configuration-options/docs.md).
+A configuration file with the required [`ServerURL`
+field](../../03.Add-ons/01.Remote-Terminal/30.Mender-shell-configuration-file/50.Mender-shell-configuration-options/docs.md#serverurl),
+set via [`MENDER_SERVER_URL`
+variable](../99.Variables/docs.md#mender_server_url), and [`User`
+field](../../03.Add-ons/01.Remote-Terminal/30.Mender-shell-configuration-file/50.Mender-shell-configuration-options/docs.md#user),
+set via [`MENDER_SHELL_USER`
+variable](../99.Variables/docs.md#mender_shell_user). These shall be set by via
```
Comment (merlin-northern) [inline 05.System-updates-Yocto-Project/05.Customize-Mender/docs.md:233]: suggested change: 'These shall be set' -> '_You can set these_'
suggested change: 'by via' -> 'via' (or something else)

File: 05.System-updates-Yocto-Project/05.Customize-Mender/docs.md [hunk 2]
```diff
@@ -222,8 +222,24 @@ IMAGE_INSTALL_append = " mender-shell"
...
+your own `.bbappend` recipe file, or via your `local.conf` file, for example:
```
Comment (merlin-northern) [inline 05.System-updates-Yocto-Project/05.Customize-Mender/docs.md:234]: I am not sure do we need comma here? "recipe file, or via your" -> "recipe file or via your"

File: 05.System-updates-Yocto-Project/99.Variables/docs.md [hunk 3]
```diff
@@ -486,7 +486,19 @@ Defaults to `${MENDER_STORAGE_DEVICE}:rootfsb` when building `.ubimg`.
 
 > Value: `https://docker.mender.io` (default)
 
-Variable to override the URL of the server for the client to connect to.
+Variable to override the URL of the server for the client to connect to. It is
```
Comment (merlin-northern) [inline 05.System-updates-Yocto-Project/99.Variables/docs.md:489]: suggested change: "Variable" -> "_A v_ariable"

File: 05.System-updates-Yocto-Project/05.Customize-Mender/docs.md (hunk 1 above)
Comment (lluiscampos) [reply 05.System-updates-Yocto-Project/05.Customize-Mender/docs.md:233]: > suggested change: 'These shall be set' -> 'You can set these'

@merlin-northern Unless you use your own `mender-shell.conf`, these are mandatory. Suggestion: "You have to set these"

File: 05.System-updates-Yocto-Project/99.Variables/docs.md [hunk 4]
```diff
@@ -486,7 +486,19 @@ Defaults to `${MENDER_STORAGE_DEVICE}:rootfsb` when building `.ubimg`.
...
+also used for add-ons configuration, where enabled.
+
+
+#### `MENDER_SHELL_USER`
+
+> Value: `nobody` (default)
+
+Variable to override the user to be used with `mender-shell`. See [Mender shell
```
Comment (merlin-northern) [inline 05.System-updates-Yocto-Project/99.Variables/docs.md:497]: suggested change: "Variable" -> "A variable"
suggested change to something along the lines of: "A variable holding a username which `mender-shell` starts the shell as."

File: 05.System-updates-Yocto-Project/99.Variables/docs.md (hunk 3 above)
Comment (lluiscampos) [reply 05.System-updates-Yocto-Project/99.Variables/docs.md:489]: "Variable to ..." is used elsewhere. I'll leave it as-is.

File: 05.System-updates-Yocto-Project/99.Variables/docs.md (hunk 4 above)
Comment (lluiscampos) [reply 05.System-updates-Yocto-Project/99.Variables/docs.md:497]: > suggested change: "Variable" -> "A variable"

It is used in other places, I'll leave it.

> suggested change to something along the lines of: "A variable holding a username which the mender-shell starts the shell as."

:+1:

File: 05.System-updates-Yocto-Project/05.Customize-Mender/docs.md (hunk 1 above)
Comment (lluiscampos) [reply 05.System-updates-Yocto-Project/05.Customize-Mender/docs.md:233]: Okey, I decided for "you ca set". Technically speaking is voluntary, but the resulting default will be useless. But okey

**Review by merlin-northern — APPROVED** [review]
a couple of suggestions. approving since I know you are going to make good use of them :) or just throw them out.

File: 05.System-updates-Yocto-Project/05.Customize-Mender/docs.md [hunk 5]
```diff
@@ -222,8 +222,24 @@ IMAGE_INSTALL_append = " mender-shell"
...
+set via [`MENDER_SHELL_USER`
+variable](../99.Variables/docs.md#mender_shell_user). You can set these in your
+own `.bbappend` recipe file or via your `local.conf` file, for example:
+
+```bash
+MENDER_SERVER_URL = "https://hosted.mender.io"
+MENDER_SHELL_USER = "root"
+```
+
+To add optional fields, or override the values for the required ones, create your own `mender-shell.conf` and
```
Comment (oleorhagen) [inline 05.System-updates-Yocto-Project/05.Customize-Mender/docs.md:241]:
```suggestion
To add optional fields, or override the values for the required ones, create your own `mender-shell.conf` and
```

File: 05.System-updates-Yocto-Project/99.Variables/docs.md [hunk 6]
```diff
@@ -486,7 +486,19 @@ Defaults to `${MENDER_STORAGE_DEVICE}:rootfsb` when building `.ubimg`.
...
+
+
+#### `MENDER_SHELL_USER`
+
+> Value: `nobody` (default)
+
+Variable holding the username which the `mender-shell` starts the shell as. See
+[Mender shell configuration
+options](../../03.Add-ons/01.Remote-Terminal/30.Mender-shell-configuration-file/50.Mender-shell-configuration-options/docs.md#user).
+Note that the default value with `meta-mender-core` is `nobody`, while with
```
Comment (oleorhagen) [inline 05.System-updates-Yocto-Project/99.Variables/docs.md:500]:
```suggestion
Note that the default value in `meta-mender-core` is `nobody`, while with
```

File: 05.System-updates-Yocto-Project/99.Variables/docs.md [hunk 7]
```diff
@@ -486,7 +486,19 @@ Defaults to `${MENDER_STORAGE_DEVICE}:rootfsb` when building `.ubimg`.
...
+`meta-mender-demo` is `root`.
```
Comment (oleorhagen) [inline 05.System-updates-Yocto-Project/99.Variables/docs.md:501]:
```suggestion
`meta-mender-demo` it is `root`.
```

File: 05.System-updates-Yocto-Project/99.Variables/docs.md [hunk 8]
```diff
@@ -486,7 +486,19 @@ Defaults to `${MENDER_STORAGE_DEVICE}:rootfsb` when building `.ubimg`.
...
-Variable to override the URL of the server for the client to connect to.
+Variable to override the URL of the server for the client to connect to. It is
+also used for add-ons configuration, where enabled.
+
+
+#### `MENDER_SHELL_USER`
+
+> Value: `nobody` (default)
+
+Variable holding the username which the `mender-shell` starts the shell as. See
```
Comment (oleorhagen) [inline 05.System-updates-Yocto-Project/99.Variables/docs.md:497]:
```suggestion
Variable holding the username which the `mender-shell` employs. See
```

**Review by oleorhagen — APPROVED** [review]

File: 05.System-updates-Yocto-Project/05.Customize-Mender/docs.md (hunk 5 above)
Comment (lluiscampos) [reply 05.System-updates-Yocto-Project/05.Customize-Mender/docs.md:241]: I can't spot the difference...

Comment (oleorhagen) [reply 05.System-updates-Yocto-Project/05.Customize-Mender/docs.md:241]: Wtf, sry. I had to refresh whilst doing the review, so something happened obvs
```suggestion
Add optional, or override required fields, with your own `mender-shell.conf` file. Then add the file to the build with...
```

**Comment by lluiscampos** [comment]
@oleorhagen See #1265

**Pull request merged**